docs/docsite/rst/dev_guide/testing.rst

```diff
@@ -83,22 +83,31 @@ Occasionally you may find your PR fails due to a reason unrelated to your change
 * a temporary issue accessing an external resource, such as a yum or git repo
 * a timeout creating a virtual machine to run the tests on
 
-If either of these issues appear to be the case, you can rerun the Azure Pipelines test by:
+If either issue appears to be the case, you can rerun the Azure Pipelines test by:
 
-* adding a comment with ``/rebuild`` (full rebuild) or ``/rebuild_failed`` (rebuild only failed CI nodes) to the PR
-* closing and re-opening the PR (full rebuild)
-* making another change to the PR and pushing to GitHub
+* adding a comment with ``/rebuild`` (full rebuild) or ``/rebuild_failed`` (rebuild only failed CI nodes) to the pull request
+* closing and re-opening the pull request (full rebuild)
+* making another change to the branch and pushing to GitHub
 
-If the issue persists, please contact us in the ``#ansible-devel`` chat channel (using Matrix at ansible.im or using IRC at `irc.libera.chat <https://libera.chat/>`_).
+If the issue persists, please contact us in the ``#ansible-devel`` chat channel (using `Matrix <https://matrix.to/#/#space:ansible.com>`_ or using IRC at `irc.libera.chat <https://libera.chat/>`_).
 
 
 How to test a PR
 ================
 
-Ideally, code should add tests that prove that the code works. That's not always possible and tests are not always comprehensive, especially when a user doesn't have access to a wide variety of platforms, or is using an API or web service. In these cases, live testing against real equipment can be more valuable than automation that runs against simulated interfaces. In any case, things should always be tested manually the first time as well.
+Ideally, the code should add tests that prove that the code works. That's not always possible and tests are not always comprehensive, especially when a user doesn't have access to a wide variety of platforms, or is using an API or web service. In these cases, live testing against real equipment can be more valuable than automation that runs against simulated interfaces. In any case, things should always be tested manually the first time as well.
 
 Thankfully, helping to test Ansible is pretty straightforward, assuming you are familiar with how Ansible works.
 
+Setup: Installing Pytest and required Pytest libraries
+------------------------------------------------------
+
+Ansible's unit testing framework leverages the pytest library. Before diving into testing, ensure you have ``pytest`` installed alongside any additional pytest libraries such as ``pytest-mock`` and ``pytest-xdist``.
+
+You can usually install pytest and most libraries using your system's package manager or using pip ``pip install pytest pytest-mock``. Refer to the documentation of your chosen option for specific installation instructions.
```

Comment:
I don't think we need to provide pip based instructions for these packages.  `--requirements`, `--venv`, or `--docker` would be recommended, as then `ansible-test` can handle the versions and constraints of the packages it requires.

Reply:
Make sense.